prow/plugins/label/label.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package label
+
+import (
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const pluginName = "label"
+
+var (
+	AllowedPrefixes    = []string{"sig", "area", "kind"}
+	labelRegex         = regexp.MustCompile(`(?m)^/label[\t\s]+(?:\S*(?:\s+|$))*`)
+	LabelNotAllowedMsg = "Label %v is not allowed to be added with the label command"
+)
+
+func init() {
+	plugins.RegisterIssueCommentHandler(pluginName, handleIssueComment)
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	IsMember(org, user string) (bool, error)
+	AddLabel(owner, repo string, number int, label string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+}
+
+func handleIssueComment(pc plugins.PluginClient, ic github.IssueCommentEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, ic)
+}
+
+func isAllowed(label string) bool {
+	for _, pref := range AllowedPrefixes {
+		if strings.HasPrefix(label, pref) {
+			return true
+		}
+	}
+	return false
+}
+
+func handle(gc githubClient, log *logrus.Entry, ic github.IssueCommentEvent) error {
+
+	commenter := ic.Comment.User.Login
+	owner := ic.Repo.Owner.Name
+	repo := ic.Repo.Name
+	number := ic.Issue.Number
+
+	member, err := gc.IsMember(repo, commenter)
+	if err != nil {
+		return fmt.Errorf("IsMember failed: %v", err)
+	}
+	if !member {
+		return gc.CreateComment(owner, repo, number, plugins.FormatResponse(ic.Comment, "Only kubernetes org members can add labels."))
+	}
+
+	match := labelRegex.FindStringSubmatch(ic.Comment.Body)
+	if match == nil || len(match) == 0 {
+		return nil
+	}
+	for _, newLabel := range strings.Split(match[0], " ")[1:] {
+		newLabel = strings.TrimSpace(newLabel)
+		if isAllowed(newLabel) {
+			gc.AddLabel(owner, repo, number, newLabel)
+		} else {
+			gc.CreateComment(owner, repo, number, fmt.Sprintf(LabelNotAllowedMsg, newLabel, number))
```

Comment:
Maybe accumulate the list of labels that are disallowed and make one comment at the end? Right now if I do `/label approve lgtm somethingelse` I'll get 3 comments, when really it can be one.